### PR TITLE
Add NIH Chest X-rays dataset.

### PR DESCRIPTION
**Pull request by MylesBartlett**
* Add NIH Chest X-rays dataset (https://www.nih.gov/news-events/news-releases/nih-clinical-center-provides-one-largest-publicly-available-chest-x-ray-datasets-scientific-community)
* Update pyproject.toml to use new group syntax.

### CODE REVIEW

File: conduit/data/datasets/vision/nih.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        Pleural_Thickening    0.030191
+        Pneumonia             0.012763
+        Pneumothorax          0.04728
+    """
+
+    ATELECTASIS = "Atelectasis"
+    CARDIOMEGALY = "Cardiomegaly"
+    CONSOLIDATION = "Consolidation"
+    EDEMA = "Edema"
+    EFFUSION = "Effusion"
```
Comment (thomkeh) [inline conduit/data/datasets/vision/nih.py:52]: are we doing ALL_CAPS enums again?

**Review by thomkeh — APPROVED** [review]

File: conduit/data/datasets/vision/nih.py (hunk 1 above)
Comment (MylesBartlett) [reply conduit/data/datasets/vision/nih.py:52]: I think the plan was always to gradually transition to using them throughout conduit as all caps is the convention for enums

Comment (MylesBartlett) [reply conduit/data/datasets/vision/nih.py:52]: However, do you think it'd be better to just remain internally consistent?

Comment (thomkeh) [reply conduit/data/datasets/vision/nih.py:52]: No, as long as you have a plan, it's fine.

**Comment by thomkeh** [comment]
hm, something about `sklearn` as a dependency?

**Comment by thomkeh** [comment]
EthicML depends on `folktables` which depends on `sklearn` (which should be `scikit-learn`).

**Comment by thomkeh** [comment]
There is an issue for it: https://github.com/zykls/folktables/issues/32

File: conduit/data/datasets/vision/nih.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    `this
+    <https://www.nih.gov/news-events/news-releases/nih-clinical-center-provides-one-largest-publicly-available-chest-x-ray-datasets-scientific-community>`__
+    Open Access paper.
+
+    The dataset can be downloaded by following the above link or from `kaggle <https://www.kaggle.com/datasets/nih-chest-xrays/data>`__
+    """
+
+    TARGET_ATTR: TypeAlias = NIHTargetAttr
+    SENS_ATTR: TypeAlias = NIHSensAttr
+    SPLIT: TypeAlias = NIHSplit
```
Comment (thomkeh) [inline conduit/data/datasets/vision/nih.py:82]: I don't think the TypeAliases should be in ALL CAPS?

Comment (MylesBartlett) [reply conduit/data/datasets/vision/nih.py:82]: 👍

**Pull request merged**